app/controllers/claims_controller.rb

```diff
@@ -121,10 +120,6 @@ def update
       check_still_teaching_params
     else
       current_claim.attributes = claim_params
-
-      # If some DQT data was missing and the user fills them manually we need
-      # to re-populate those answers which depend on the manually-entered answer
-      set_dqt_data_as_answers if current_claim.qualifications_details_check
```

Comment:
🤔 is this not needed when DQT returns incomplete data and the user is directed to enter it manually?

Reply:
Best I can tell I don't think this actually does anything. Quite hard to follow what with the answer reseting and all but, on master, if DQT returns missing data, eg it returns a payload like
```
      {
        trn: 123456,
        ni_number: "AB123123A",
        name: "Rick Sanchez",
        dob: "66-06-06T00:00:00",
        active_alert: false,
        state: 0,
        state_name: "Active",
}
```
then the journey just kind of breaks. It assumes the `qualification` (ultimately set by `claim.dqt_teacher_record&.route_into_teaching`) is `:postgraduate_itt`, so as `claim.dqt_teacher_record&.route_into_teaching` is returning a value, `qualification` is removed from the slug sequence. Itt year ends up being in the slug sequence as it's not returned from DQT, however as we set `itt_academic_year` on the `itt-year` page,  that resets `eligible_itt_subject`, but the slug sequence is checking for `claim.dqt_teacher_record&.eligible_itt_subject_for_claim` which returns `none_of_the_above` if no value is set ¹, so `eligible-itt-subject` is removed from the slug sequence. (apologies, this was mostly a brain dump for me!)

I'll try and write some feature specs to cover this but I'm not sure if these fixes want breaking out into a separate ticket.

https://github.com/DFE-Digital/claim-additional-payments-for-teaching/assets/9936028/6885f86a-d830-4b21-a6aa-46b6d040a70f

1. For the ECP journey, for LUP it returns `nil`

Reply:
@slorek No, reset dependent attributes doesn't kick in if there's data from DQT
https://github.com/DFE-Digital/claim-additional-payments-for-teaching/blob/901e46fa0f77da29da162e65e382dd8ffbc4ec23/app/models/policies/levelling_up_premium_payments/eligibility.rb#L93
https://github.com/DFE-Digital/claim-additional-payments-for-teaching/blob/901e46fa0f77da29da162e65e382dd8ffbc4ec23/app/models/policies/early_career_payments/eligibility.rb#L124

Though there's quite a few bugs if DQT only returns partial data, eg:

https://github.com/DFE-Digital/claim-additional-payments-for-teaching/assets/9936028/d74da0c4-0824-46be-94f3-8c444239bb08

so we'll probably want to do something like only show `qualification-details` step if DQT returns sufficient data (not sure how much of a problem this is IRL)

Reply:
There's also some assumptions made about missing DQT data I'm not sure about, eg for an ECP claim, if DQT is missing `initial_teacher_training.qualification` then it assumes the qualification is `:postgraduate_itt`, ultimately due to the `nil` [here](https://github.com/DFE-Digital/claim-additional-payments-for-teaching/blob/901e46fa0f77da29da162e65e382dd8ffbc4ec23/lib/dqt/matchers/general.rb#L6). There's a few other things like that such as a missing `qualified_teacher_status.qts_date` setting `eligible_itt_subject` on the claim's eligibility to `none_of_the_above` and as the value isn't nil the teacher isn't presented with the option to change it. I don't have enough business context to say if these are bugs but they do seem a bit odd!

Reply:
Thanks @rjlynch - I think it would be worth creating a ticket for each of these to investigate further.

Reply:
Done 👍 Only made one ticket as all the issues manifest in a single scenario https://dfedigital.atlassian.net/browse/CAPT-1624